### PR TITLE
feat: Redesigning of the home page

### LINKED ISSUE #480: [Advanced]: Update the "What is Hiero Section?"
### 🧑‍🔬 Advanced Issue

Welcome! This is an **[Advanced Issue](https://github.com/issues?q=is%3Aopen%20is%3Aissue%20org%3Ahiero-ledger%20archived%3Afalse%20no%3Aassignee%20label%3A%22advanced%22%20repo%3Ahiero-ledger%2Fhiero-website)** touching core website architecture.

It is designed for expert contributors who have demonstrated deep architectural understanding and a proven track record of high-quality contributions.

### 🐞 Problem Description

Our redesign introduces a new format for our "What is Hiero Section"

We want to implement a design similar to what is shown here. 
https://www.figma.com/proto/t0ysdxkgBEBXHH5JozRDs9/Hiero?node-id=1-5&p=f&viewport=293%2C268%2C0.03&t=Oipgqgcz2V79RNXl-1&scaling=

### 🛠️ Implementation Notes

Implement a design that matches the new scope from our redesign file linked above. 

* Remove the "what is hiero section" title and description. 
* Remove sub text for our larger headings: (i.e., "Do not sacrifice performance for fairness or security. As businesses and networks expand rapidly, waiting for transactions to finalize no longer meets your needs. The hashgraph algorithm is fast." )
* Stack our major headings (i.e Hiero Is leaderless) using our animation. Our goal is to make this a 3D animation. 

See example below: 
https://hedera.com/

PUSH WORK TO websiteRedesign BRANCH

Markdown Specification: 
https://github.com/hiero-ledger/hiero-website/blob/main/docs/design-specs.md

**Blog & Content Workflow:**
- [Writing Blog Posts](https://github.com/hiero-ledger/hiero-website/blob/main/docs/blogs.md) — covers the current posting flow, front matter format, and content conventions

**Project Conventions:**
- Components live in `src/components/` — follow existing patterns for props, naming, and file structure
- Pages are under `src/app/` using the Next.js App Router
- Content (blog posts, etc.) lives in `content/`
- Styling uses [Tailwind CSS](https://tailwindcss.com/docs)

### 🧠 Advanced Contributors — Prerequisites & Expectations

> [!CAUTION]
> **Advanced issues are the highest-risk work in this project. We will reject PRs that do not meet these standards.**

### 🏁 Concrete Prerequisites
- **Advanced Web Development:** Proficient with React, Next.js, and TypeScript.
- **Expertise:** Deep architectural understanding of the website's component structure, routing, and build pipeline.
- **Proven History:** Successfully completed **≥ 10 intermediate issues** in this repo.
- **Consistency:** **≥ 3–4 months** of active, human-led contributions to this project.

> [!NOTE]
> **CI/CD Exception:** For issues focused on **GitHub Actions / Workflows**, the repo-specific thresholds above may be waived if the contributor demonstrates advanced-level proficiency in CI/CD.

### ⚠️ AI Usage Policy

- Using AI to generate code for Advanced issues is **strictly discouraged**
- AI may be used to help explain file relationships, but cannot be the main source of research.
- Submitting AI-generated or unvalidated code is grounds for **immediate closure**

### ⏱️ Timeline & Workflow
- **Typical time:** ~1 month / ~50 hours.
- 🔴 Completing an advanced issue in 1–3 days is a **red flag** and will likely be rejected.
- **Suggested:** Post your proposed architectural approach as a comment and wait for explicit maintainer approval **before writing any code.**

### 🔬 Technical Domains

- [ ] **Advanced React / Next.js** (Server Components, App Router, dynamic routing, middleware)
- [ ] **Component Architecture** (design systems, reusable patterns, composition)
- [ ] **Performance & Optimization** (Core Web Vitals, image optimization, code splitting, caching)
- [ ] **Accessibility** (WCAG compliance, screen reader support, keyboard navigation)
- [ ] **Build & Deployment** (Next.js config, CI/CD, hosting/deployment integration)
- [ ] **Testing** (unit, integration, E2E, accessibility testing, visual regression)

### 🛡️ Quality & Review Standards

Advanced PRs must be **"safe, maintainable, architecturally sound, and production-ready."**

1. **Architectural Fit:** The solution must fit naturally into the existing website architecture.
2. **Security & Correctness:** Evaluate all logic for injection risks, XSS vulnerabilities, or state corruption issues. 
3. **Maintainability:** Code must be short and clear enough for any other maintainer to debug without your assistance. 
4. **Backward Compatibility:** Existing pages, routes, and components must not break. If a breaking change is required, it must be explicitly managed.
5. **Comprehensive Testing:** Must include unit and integration tests covering all new logic paths, edge cases, and failure modes. AI-generated tests based on AI-generated code are grounds for immediate rejection.

### ✅ PR Quality Checklist

Before opening your PR, the contributor must confirm:
- [ ] I have spent the majority of my time researching the problem and solution space extensively, including reviewing relevant code, documentation, and external resources.
- [ ] I understand the system-wide impact of these changes on affected modules and performance.
- [ ] The design fits with the current website's architectural approaches.
- [ ] I have tested my changes extensively across browsers and screen sizes.
- [ ] Every line of code is personally understood and explainable.

### 📚 Resources & Support

**References:**
- [Next.js Documentation](https://nextjs.org/docs)
- [React Documentation](https://react.dev)
- [Tailwind CSS Documentation](https://tailwindcss.com/docs)
- [Vitest Documentation](https://vitest.dev)
- [Writing Blog Posts](https://github.com/hiero-ledger/hiero-website/blob/main/docs/blogs.md)

**🆘 Stuck?**
- [Community Calls](https://zoom-lfx.platform.linuxfoundation.org/meetings/hiero?view=week)
- [Discord](https://github.com/hiero-ledger/sdk-collaboration-hub/blob/main/guides/issue-progression/for-developers/discord.md)

### PR DESCRIPTION
**Pull request by danielmarv**
# Description
This pull request significantly refactors and expands the `HeroSection` component, introducing two new visual subcomponents (`ConsensusLattice` and `HeroTraces`), updating the structure and data model, and adding comprehensive tests to ensure correct rendering and accessibility. The changes modernize the hero section with interactive and visually rich elements, improve accessibility, and provide robust test coverage.

**Key improvements and new features:**

### Hero Section Redesign

* Refactored `HeroSection` to use a new data model with `eyebrow`, `heading`, `lede`, `actions`, and `facts`, and updated its structure to include the new `ConsensusLattice` and `HeroTraces` visualizations. Accessibility has been improved by using semantic elements and ARIA attributes. (`src/components/HeroSection/index.tsx`)

### New Visual Components

* Added `ConsensusLattice`, a dynamic, interactive 3D visualization of the consensus protocol, including pointer parallax effects and precise geometry calculations to ensure consistency between server and client rendering. (`src/components/HeroSection/ConsensusLattice.tsx`)
* Added `HeroTraces`, a static SVG network visualization representing gossip traces, generated deterministically for server/client consistency and accessibility. (`src/components/HeroSection/HeroTraces.tsx`)

### Testing Enhancements

* Added comprehensive tests for `ConsensusLattice` and `HeroTraces` to verify correct rendering, geometry, accessibility, and deterministic output. (`src/components/HeroSection/__tests__/ConsensusLattice.test.tsx`, `src/components/HeroSection/__tests__/HeroTraces.test.tsx`) [[1]](diffhunk://#diff-9e3974003e39063e61fc6ac389eafefc190c491088cfb57d2c20715dabc1e1d8R1-R55) [[2]](diffhunk://#diff-b7b6965d2add36bd08aa91c62ed716557c4c5ddc80dec7d888553ae15cf4af93R1-R39)
* Updated `HeroSection` tests to mock visual components for focused snapshot and behavior testing, and expanded test cases to cover all new behaviors and accessibility requirements. (`src/components/HeroSection/__tests__/HeroSection.test.tsx`, `src/components/HeroSection/__tests__/__snapshots__/HeroSection.test.tsx.snap`) [[1]](diffhunk://#diff-2d47d9ac000482d09dd5b481ec36cf4eda25bd02e2965a5a1a61eb9bbc98b2aaL1-L21) [[2]](diffhunk://#diff-00438b9049891de76707fdba468b3e1d38435ff38caa431167fad7fa8fa648beL3-R102)

### Minor Improvements

* Allowed `RichText` to render as an `h2` element, increasing its flexibility. (`src/components/RichText/index.tsx`)

## Changes Made
- [ ] Added...
- [ ] Modified...
- [ ] Fixed...

## Related Issues

Fixes: #480 
Fixes: #479 
Fixes: #477 

## Checklist
- [ ] Tests added/updated
- [ ] Documentation updated
- [ ] Linting passes
- [ ] Branch up-to-date with main

### CODE REVIEW

**Review by aceppaluni — CHANGES_REQUESTED** [review]
Is this refactoring the homepage?

File: src/app/globals.css [hunk 1]
```diff
@@ -56,6 +56,582 @@
...
+
+  .hero-section-actions {
+    @apply flex flex-wrap gap-3;
+  }
+
+  .hero-section-action {
+    @apply inline-flex h-12 items-center gap-2 rounded-full px-6.5;
+    @apply border border-transparent text-base font-medium leading-none no-underline;
+    @apply transition-[background-color,border-color,translate] duration-200 ease-out;
+    @apply hover:-translate-y-px;
```
[the remark this answers is not in the document]
Comment (exploreriii) [reply src/app/globals.css]: the section's entries fade in without the intended rise and then sit permanently 20px below their position

**Comment by danielmarv** [comment]
cc @exploreriii @aceppaluni

**Comment by danielmarv** [comment]
> Is this refactoring the homepage?

yes now it is

File: src/components/GossipField/index.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+const SEEDS: Record<GossipFieldPlacement, number> = {
+  principles: 58201774,
+  issues: 20260901,
+  calls: 41720433,
+  repos: 77310219,
+  quotes: 13480967,
+};
+
+/** Built once per placement, at module scope, for the reason above. */
```
Comment (aceppaluni) [inline src/components/GossipField/index.tsx]: Am confused about the reasoning here?

Should we add something like this 
```markdown
/**
 * Generate each field once when the module is loaded rather than on every
 * render. The fields are deterministic from their seeds, so keeping the
 * generated data at module scope avoids repeating the relatively expensive
 * layout calculation while preserving stable visuals for each placement.
 */
```

File: src/components/GossipField/index.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+const FIELDS = Object.fromEntries(
+  Object.entries(SEEDS).map(([placement, seed]) => [
```
Comment (aceppaluni) [inline src/components/GossipField/index.tsx]: Object.fromEntries(... ) as Record<...> weakens type safety. 

Do you feel as though this is a concern?

File: src/components/HeroSection/HieroMarkScene.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+  "--hero-art-rotate-y": ({ x }) => `${(x * 5).toFixed(2)}deg`,
+  "--hero-traces-x": ({ x }) => `${(x * -13).toFixed(2)}px`,
+  "--hero-traces-y": ({ y, scroll }) =>
+    `${(y * -9 - scroll * 34).toFixed(2)}px`,
+  "--hero-halo-x": ({ x }) => `${(x * 8).toFixed(2)}px`,
+  "--hero-halo-y": ({ y, scroll }) => `${(y * 6 - scroll * 18).toFixed(2)}px`,
+  "--hero-scroll-progress": ({ scroll }) => scroll.toFixed(3),
+};
+
+const MARK = (
```
Comment (aceppaluni) [inline src/components/HeroSection/HieroMarkScene.tsx]: Do you think this should this be made into a reusable component?

File: src/components/QuotesCarousel/index.tsx [hunk 5]
```diff
@@ -1,120 +1,188 @@
...
 
+export const AUTO_ADVANCE_MS = 10_000;
+
 export default function QuotesCarousel({ data }: QuotesCarouselProps) {
-  const swiperRef = useRef<SwiperRef>(null);
+  const [activeIndex, setActiveIndex] = useState(0);
+  const [direction, setDirection] = useState<"next" | "previous">("next");
+  const [autoPlayEnabled, setAutoPlayEnabled] = useState(true);
+  const [interactionPaused, setInteractionPaused] = useState(false);
+  const activeQuote = data.quotes[activeIndex];
```
Comment (aceppaluni) [inline src/components/QuotesCarousel/index.tsx:32]: If data.quotes is empty for any reason, these may all crash. Should we add a conditional?

```
if (data.quotes.length === 0) {
  return null;
}
```

File: src/components/QuotesCarousel/index.tsx [hunk 6]
```diff
@@ -1,120 +1,188 @@
...
+              </blockquote>
+            </article>
+          </div>
+
+          {data.quotes.length > 1 ? (
+            <div className="quotes-carousel-controls">
+              <div className="quotes-carousel-status">
+                <p>Community perspectives</p>
+                <div
+                  className="quotes-carousel-progress"
```
Comment (aceppaluni) [inline src/components/QuotesCarousel/index.tsx]: This 
```javascript
<div
  className="quotes-carousel-progress"
  aria-label={`Quote ${activeIndex + 1} of ${data.quotes.length}`}
>
```
may not be announced as we expect because it's just a div. Should we either make the current position part of the carousel's accessible state or add a visually hidden status?

Something like
```javascript
<p className="sr-only" aria-live="polite">
  Quote {activeIndex + 1} of {data.quotes.length}
</p>
```
Could be useful for keyboard users who press Previous/Next.

**Review by aceppaluni — CHANGES_REQUESTED** [review]
My additional question is how this would look on mobile. Are you able to provide a screenshot for the mobile preview?

File: src/components/GossipField/index.tsx (hunk 2 above)
Comment (exploreriii) [reply src/components/GossipField/index.tsx]: @aceppaluni click F12 and at the top, typically in the middle, click device. There you can change your viewport to many different types of mobile devices. So, run the redesign locally and F12 to see how it renders on lots of the devices and what happens if you move things too - will be more accurate than the screenshot.

**Comment by danielmarv** [comment]
> My additional question is how this would look on mobile. Are you able to provide a screenshot for the mobile preview?

for you you inspect locally coz it would be better to do that instead of providing screenshots

**Review by hendrikebbers — APPROVED** [review]
